src/main/java/com/ap/todo/interfaces/mapper/UpdateTodoMapper.java

```diff
@@ -34,8 +34,8 @@ protected void afterMappingToCommand(
             @MappingTarget final UpdateTodoCommand.UpdateTodoCommandBuilder targetBuilder,
             UpdateTodoReqDto reqDto) {
         if(Objects.nonNull(reqDto.getExecutionDate())) {
-//            LocalDateTime convertedExecutionDate = LocalDateTime.parse(reqDto.getExecutionDate(), DateTimeFormatter.ofPattern("yyyyMMddHHmmss"));
-//            targetBuilder.executionDate(convertedExecutionDate);
+            LocalDateTime convertedExecutionDate = LocalDateTime.parse(reqDto.getExecutionDate(), DateTimeFormatter.ofPattern("yyyyMMddHHmmss"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

### DateTimeFormatter.ofPattern [](|67999369-6dcc-4893-b659-89de93884ecc|)
#### Note: “I18N doesn’t apply because this use case is internal” is a misconception. For formatting that isn’t user-facing: Set Locale to Locale.ROOT, set TimeZone to UTC, and check if your format is [predefined in DateTimeFormatter](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html).

Picture strings (for example, `MM-dd-yyyy` and `HH:mm:ss`) that you specify to represent the date won’t be appropriate for all locales, and should not be hardcoded. Different locales use different date and time formats, positioning, punctuation, and even numbers themselves. So, the developer or a translator has to adapt the string for each locale&#39;s specific needs. Otherwise, the results could look strange or wrong to users in another language or locale.

Also, DateTimeFormatter.ofPattern shouldn’t be used to fix issues with SimpleDateFormat, since the two produce similar results. The [Internationalization channel on Broadcast](https://broadcast.amazon.com/channels/28439) has two videos on the challenges of using SimpleDateFormat for [User-displayed customer experience (CX)](https://broadcast.amazon.com/videos/310354) and [Application-internal timestamps and application-specific/application-defined values](https://broadcast.amazon.com/videos/307513).

### Suggested solutions:
#### For user-facing date and time:

1. **[PREFERRED]** [MessageFormat](https://w.amazon.com/bin/view/I18N/I18N_Best_Practices/I18N_Imports/) supports &#34;skeletons&#34; that can automatically select the right pattern for the customer&#39;s locale. To learn more, see [How to format dates](https://w.amazon.com/bin/view/INTech/docs/formatting-values/how-to-format-dates/) and the [MessageFormat Cheat Sheet](https://w.amazon.com/bin/view/INTech/docs/formatting-values/messageformat-tldr/).
    ```
    MessageFormat fmt = new MessageFormat(pattern, locale); 
    Object[] args = { new Date() }; 
    return fmt.format(args);
    ```
If you want to get the string translated, create a resource file for the date and time pattern. Use Panther to get the date and time pattern string translated. MessageFormat is built into [Panther](https://w.amazon.com/bin/view/INTech/Panther). See the [DateTimeFormatter.ofPattern]({0}) wiki.

2. Consider using the ICU4J library `DateFormat.getPatternInstance` method (which uses a date skeleton string) instead of the stock Java `DateFormat` class. Skeleton-based date and time formats adapt to the locale automatically.
    ```
    import com.ibm.icu.text.DateFormat; // note different import!
    DateFormat df = DateFormat.getPatternInstance(&#34;jmsz&#34;, pageLocale);
    ```

#### For application-internal timestamps:
1. Use a predefined ISO 8601 format like `ISO_INSTANT` (which looks like `2011-12-03T10:15:30Z`) from [DateTimeFormatter](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html).
    ```
    String now = DateTimeFormatter.ISO_INSTANT.format(new Date().toInstant());
    ```

2. If your format is not *[predefined in DateTimeFormatter](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html)* and you *must* use a custom format:
    - Make sure you pass in the Locale (for non-user-facing, internal-processing usage, use `Locale.ROOT` )
    - Set the timezone (for example, `UTC`), if you want to ensure it is a certain timezone.
    
    Example:
    ```
    // your custom pattern goes here
    final String somePattern = &#34;yyyy MM dd HH mm ss z&#34;; // $NON-NLS-L$ Custom hard-coded pattern
    
    DateTimeFormatter myFormatter = 
    DateTimeFormatter.ofPattern(somePattern, Locale.ROOT)  // $NON-NLS-L$ custom pattern, invariant locale
            // only add this line if you want to ensure UTC (or some other time zone)
            .withZone(ZoneId.of(&#34;UTC&#34;)); // $NON-NLS-L$ forcing UTC
    
    myFormatter.format(/* some value */);
    ```